idaes/power_generation/flowsheets/supercritical_steam_cycle/supercritical_steam_cycle.py

```diff
@@ -224,6 +224,10 @@ def mixer_pressure_constraint(b, t):
     @m.fs.hotwell.Constraint(m.fs.time)
     def mixer_pressure_constraint(b, t):
         return b.condensate_state[t].pressure == b.mixed_state[t].pressure
+    # PYLINT-TODO check if the "function-redefined" pylint error can be addressed
```

Comment:
This is a tricky one. Pylint is correctly identifying that the methods are being redefined, although that does not affect code functionality as they are being used in different namespaces (which pylint cannot easily see). This could easily be fixed by changing the name of the method (e.g. appending a prefix or suffix for the unit in question), but we would need to make sure that other parts of the code are not looking for these constraints by name.

Alternatively, using call syntax would work fine, and avoid any issues with changing the name of the constraint. The rules are sufficiently different to preclude writing a common method for all, but each constraint could define a new method with a unique name and not affect the code.

Reply:
Short answer - I suggest refactoring the code, which should not be hard.

Reply:
Thanks for the explanation, it was useful for me to understand the context a bit better. On second thought, my personal preference would be to add the pylint directive (leaving the code as it is) rather than making changes to the code:

- I initially thought that the functionality (i.e. the inner function's body) was identical in all cases, but as you point out, this is not the case
- This particular portion of the code doesn't seem to be covered by unit tests (might be by component/integration tests)
- I peeked at the source code for `Constraint` in `pyomo.core.base.constraint` and, unless the implementation used in this part of the IDAES code is significantly simpler, I think I've seen enough ~dark magic~ metaprogramming to make me hesitant to jump in and make the changes without more research or ~amulets~ tests
- In general, I would argue that the type of error that pylint is flagging here does not apply in this particular situation, as the decorator is applied directly to the function definition; therefore, I'd lean more into considering this a false positive

Reply:
I agree that it is a false positive, but I could also see it confusing someone reading the code as well. For me, I am somewhat hesitant to add pylint directive unless absolutely necessary, in case this ends up masking real problems in the future.

That said, this is not something I would ask you to fix - if we were going to change the model code I would ask the powerplant team to do it.